dataplane/docs/dataplaneforsonic.md

```diff
@@ -0,0 +1,61 @@
+│  │  │   │                   │               │  │  │                  │  │
+│  │  │   │                   │               │  │  │                  │  │
+│  │  │   │ syncd             │               │  │  │                  │  │
+│  │  │   └───────────────────┘            ┌──┴──┴──┴──┐               │  │
+│  │  │                                    │ Ethernet4 │               │  │
+│  │  │                                    └──┬──┬──┬──┘               │  │
+│  │  │ VM                                    │  │  │                  │  │
+│  │  └───────────────────────────────────────┘  │  │                  │  │
+│  │                                             │  │                  │  │
+│  │  SONiC VM Runner                            │  │ Dataplane        │  │
+│  └─────────────────────────────────────────────┘  └──────────────────┘  │
+│                                                                         │
+│                                                                         │
+│                                                    ┌──────┐  ┌──────┐   │
+│ pod                                                │ eth1 │  │ eth2 │   │
+└────────────────────────────────────────────────────┴──────┴──┴──────┴───┘
+```
+
+## libsai
+
+The dataplane libsai is shared object that implements the SAI API. It composed of almost entirely autogenerated code that converts the C structs to protobuf struct. The code is generated by the apigen Go library.
+
+In addition to SAI implementation, the CPU Port manager runs as a another thread in the library. The CPU port manager is responible for handling IPs address assign the interfaces inside the VM.
+
+The process for handling packets destined for the host is:
+
+1. A hostif is created called Ethernet0 with OID 3.
+2. saiserver receives the request and sets up the dataplane
+3. saiserver notifies the CPU manager that a hostif is created
+4. the user assignes the IP 192.168.1.100/24 to the Ethernet0
+5. syncd created a route with prefix 192.168.1.100/32 and next hop the CPU port ID.
+6. CPU port MGR reacts to new IP (netlink message) and an entry to a dataplane table:  
+   1. If output port is CPU and IP DST header is 192.168.1.100 then output the Ethernet0 device
+
+The CPU port manager also handles [SAI GNETLINK Hostif](https://github.com/opencomputeproject/SAI/blob/master/inc/saihostif.h#L832). These are special hostif types that use the [generic netlink](https://wiki.linuxfoundation.org/networking/generic_netlink_howto) to received packets from the CPU port. Since this API is local the kernel, the packets must be written from inside the VM.
+
+## Dataplane Container
+
+The dataplane container runs the saiserver and forwarding packets between the veths present in the pod. Since all containers in a Kubernetes pod share the same namespace, the dataplane container communicates with VM using the TAP interfaces. These TAP interface are the hostifs that appear inside the VM.
```

Comment:
How many physical devices emulated in the VM? 

Are the following use cases valid?
* Use multiple PODs for a given topology?
* Run multiple SONiC VM Runners in a POD for a given topology?
* Run multiple VMs in a SONiC VM Runner for a given topology?

Reply:
pod == 1 device
* yes
* no
* no